### ISSUE TITLE
Pivot project type: Provide infrastructure to detected active tab changes

### ISSUE THREAD

**Issue by dgomezc** [Microsoft/WindowsTemplateStudio#1584]
In relation with #1504, there would be some scenarios using a Pivot project type where it is really interesting to detect when active tab changes.

An approach to have this infrastructure would be:

- Suscribe to SelectionChanged event in Pivot page and detect the old active pages and new active pages (selection changed event args return two list).

```csharp
private void Pivot_SelectionChanged(object sender, SelectionChangedEventArgs e)
{
    var selectedPages = e.GetSelectedPages();
    OnPagesSelected(selectedPages);

    var unselectedPages = e.GetUnselectedPages();
    OnPagesUnselected(unselectedPages);
}
```

- For each page list, create method to detect page type and execute required actions.

```csharp
private void OnPagesSelected(IEnumerable<Page> pages)
{
    foreach (Page page in pages)
    {
        switch (page)
        {
            case PageX                
                ...   //do actions
                break;

            default:
                break;
        }
    }
}

private void OnPagesUnselected(IEnumerable<Page> pages)
{
    foreach (Page page in pages)
    {
        switch (page)
        {
            case PageX                
                ...   //do actions
                break;

            default:
                break;
        }
    }
}
```

**Comment by crutkas** [comment]
this feels like we're doing too much

**Comment by crutkas** [comment]
i would point to docs.microsoft.com or a github doc page from us, not include in template

**Comment by ralarcon** [comment]
Makes sense, we can create the page explaining a way to solve it and mention it in the template...

**Comment by crutkas** [comment]
as a fix for #1504, this fix be added only for tabbed media player.  That i'm fine with as it is relevant to that scenario.

**Comment by ralarcon** [comment]
Our thoughs here were: What happen if more than one page wants to execute some action when is selected or unselected? And is because of that we thougth to have this basic infrastructure related to the pivot (and able to be handled using "post-actions" to add the required code when a page needs it).

Other options we can evaluate are:
1. Create a kind of "Navigation Service" for Pivot projects. This service would handle the "subscription" of pages to the selected / unselected, and other staff... this would be "too much" too, in my opinion.
2. Follow the strategy of "alone" pages (each page solves its needs): In this case we can include code in the Media player to "find" the parent Pivot and get subscribed to its SelectionChange event. Implement in the handler the actions / code needed depending on the framework used.
3. Similiar to previous point but adding a property to the App class to "publicly" allow to access the Pivot from any page without having to find it.

Point 2 or 3 are more isolated and this stragety can be used when a certain pages requires it.

@crutkas what strategy you thing is better. For me, poing 2 or 3 wins in simplicity and currently we do not have other pages requiring this kind of actions.

**Comment by mrlacey** [comment]
The premise behind this question suggests to me that this falls too close to functionality that would be better placed inside a specific framework, rather the simple code that everyone needs and should be inside a WTS template. There's a delicate balance to be found with such functionality but this feels to me to be way beyond what WTS should include for everyone. 

As discussed above the fix for the scenario in #1504 should be addressed on its own and not be solved with the provision of new infrastructure.

As no general work is required for this issue going to close it and leave the work to be done/tracked on the specific issue.
